src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -201,53 +212,55 @@ class UnixEvent
       m_state(initialState)
     {
         int st = pthread_mutex_init(&m_mutex, NULL);
-        ASSERT(st == NULL);
+        FATAL_ASSERT(st == 0, "Failed to initialize UnixEvent mutex");
```

Comment:
I have used the fail fast, since failure to create the event is non-recoverable. Or can you see a case where we could recover from the failure?